### PR TITLE
F' v2.0.0 update support

### PR DESCRIPTION
**Pull request by r9-pena**
Updated documents to support updates from v2.0.0.
Updated F' sub-module to point to nasa/fprime repo

### CODE REVIEW

File: GpsApp/Gps/GpsComponentImpl.cpp [hunk 1]
```diff
@@ -21,10 +21,16 @@ namespace GpsApp {
   // ----------------------------------------------------------------------
 
   GpsComponentImpl ::
+#if FW_OBJECT_NAMES == 1
```
Comment (Joshua-Anderson) [inline GpsApp/Gps/GpsComponentImpl.cpp:24]: This FW_OBJECT_NAMES is a pattern we decided to remove (though it looks like we forgot to update the GPS app tutorial!).

Instead we always provide a component name but substitute it with an empty string  if object names are disabled. You can see some example of this in F' Ref app.

File: GpsApp/Gps/GpsComponentImpl.hpp [hunk 2]
```diff
@@ -48,7 +48,11 @@ namespace GpsApp {
       //! Construct object Gps
       //!
       GpsComponentImpl(
+#if FW_OBJECT_NAMES == 1
```
Comment (Joshua-Anderson) [inline GpsApp/Gps/GpsComponentImpl.hpp:51]: Remove FW_OBJECT_NAMES

File: GpsApp/Top/Topology.cpp [hunk 3]
```diff
@@ -199,37 +241,52 @@ void constructApp(int port_number, char* hostname, char* device) {
...
     if (hostname != NULL && port_number != 0) {
-        socketIpDriver.startSocketTask(100, 10 * 1024, hostname, port_number);
+        Fw::EightyCharString name("ReceiveTask");
+        // Uplink is configured for receive so a socket task is started
+        comm.configure(hostname, port_number);
+        comm.startSocketTask(name, 100, 10 * 1024);
     }
+    return false;
 }
+/*
```
Comment (Joshua-Anderson) [inline GpsApp/Top/Topology.cpp:289]: I think you can safely remove this code - it's been replaced by the code you added to main.cpp

Does the tutorial reflect these changes?

File: GpsApp/Top/Topology.cpp [hunk 4]
```diff
@@ -256,18 +311,40 @@ void runcycles(NATIVE_INT_TYPE cycles) {
     }
 
 }
-
+*/
 void exitTasks(void) {
     rateGroup1Comp.exit();
     rateGroup2Comp.exit();
     rateGroup3Comp.exit();
+    //blockDrv.exit();
```
Comment (Joshua-Anderson) [inline GpsApp/Top/Topology.cpp:319]: Is there a reason why blockDrv and pingRcvr's exit functions are commented out?

File: GpsApp/build-artifacts/Linux/dict/channels/blockDrv_channels.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (Joshua-Anderson) [inline GpsApp/build-artifacts/Linux/dict/channels/blockDrv_channels.html:1]: Make sure to add build-artifacts to the .gitignore file!

**Review by Joshua-Anderson — COMMENTED** [review]
Great work! I added a few minor notes, but everything looks good overall!

File: GpsApp/Top/Topology.cpp (hunk 4 above)
Comment (r9-pena) [reply GpsApp/Top/Topology.cpp:319]: The blockDrv was added back. The pingRcvr is not needed for the GpsApp. Would you like me to remove them completely or add them back?

Comment (Joshua-Anderson) [reply GpsApp/Top/Topology.cpp:319]: You're right, pingRcvr isn't needed. Can you entirely remove it?

File: GpsApp/Top/Topology.cpp [hunk 6]
```diff
@@ -128,66 +131,105 @@ void constructApp(int port_number, char* hostname, char* device) {
...
     fileUplink.init(30, 0);
     fileDownlink.init(30, 0);
+    fileDownlink.configure(1000, 1000, 1000, 10);
+    fileManager.init(30, 0);
     fileUplinkBufferManager.init(0);
-    fileDownlinkBufferManager.init(1);
     fatalAdapter.init(0);
     fatalHandler.init(0);
     health.init(25,0);
+    //pingRcvr.init(10);
```
Comment (Joshua-Anderson) [inline GpsApp/Top/Topology.cpp:172]: Could you possibly remove the other commented out references to `pingRcvr` and other commented out code in the topology file?

**Review by Joshua-Anderson — APPROVED** [review]
🚀

File: GpsApp/Top/Main.cpp [hunk 7]
```diff
@@ -18,16 +18,38 @@ static void sighandler(int signum) {
...
-    char* hostname = NULL;
-    char* device = (char*)"/dev/ttyUSB0";
+    U32 port_number = 0; // Invalid port number forced
+    I32 option;
+    char *hostname;
+    char *device;
+    option = 0;
+    hostname = NULL;
+    device = (char*)"/dev/ttyUSB0";
+    bool dump = false;
```
Comment (Joshua-Anderson) [inline GpsApp/Top/Main.cpp:47]: One last tiny thing I noticed doing a quick local test, I got a compiler warning that `dump` was unused... do you mind removing the variable?

**Pull request merged**